File: frameworks/python/sanic/Client/Kiota/parameters/parameters_request_builder.py
from __future__ import annotations
from collections.abc import Callable
from kiota_abstractions.base_request_builder import BaseRequestBuilder
from kiota_abstractions.get_path_parameters import get_path_parameters
from kiota_abstractions.request_adapter import RequestAdapter
from typing import Any, Optional, TYPE_CHECKING, Union

if TYPE_CHECKING:
    from .item.with_one_item_request_builder import WithOneItemRequestBuilder
    from .static.static_request_builder import StaticRequestBuilder

class ParametersRequestBuilder(BaseRequestBuilder):
    """
    Builds and executes requests for operations under /parameters
    """
    def __init__(self,request_adapter: RequestAdapter, path_parameters: Union[str, dict[str, Any]]) -> None:
        """
        Instantiates a new ParametersRequestBuilder and sets the default values.
        param path_parameters: The raw url or the url-template parameters for the request.
        param request_adapter: The request adapter to use to execute the requests.
        Returns: None
        """
        super().__init__(request_adapter, "{+baseurl}/parameters", path_parameters)
    
    def by_one(self,one: int) -> WithOneItemRequestBuilder:
        """
        Gets an item from the Kiota.parameters.item collection
        param one: Unique identifier of the item
        Returns: WithOneItemRequestBuilder
        """
        if one is None:
            raise TypeError("one cannot be null.")
        from .item.with_one_item_request_builder import WithOneItemRequestBuilder

        url_tpl_params = get_path_parameters(self.path_parameters)
        url_tpl_params["one"] = one
        return WithOneItemRequestBuilder(self.request_adapter, url_tpl_params)
    
    @property
    def static(self) -> StaticRequestBuilder:
        """
        The static property
        """
        from .static.static_request_builder import StaticRequestBuilder

        return StaticRequestBuilder(self.request_adapter, self.path_parameters)
    

File: frameworks/java/ktor/Client/Kiota/client/kiota/parameters/ParametersRequestBuilder.java
package client.kiota.parameters;

import client.kiota.parameters.item.WithOneItemRequestBuilder;
import client.kiota.parameters.staticescaped.StaticRequestBuilder;
import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import java.util.HashMap;
import java.util.Objects;
/**
 * Builds and executes requests for operations under /parameters
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class ParametersRequestBuilder extends BaseRequestBuilder {
    /**
     * The static property
     * @return a {@link StaticRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public StaticRequestBuilder staticEscaped() {
        return new StaticRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * Gets an item from the client.kiota.parameters.item collection
     * @param one Unique identifier of the item
     * @return a {@link WithOneItemRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public WithOneItemRequestBuilder byOne(@jakarta.annotation.Nonnull final String one) {
        Objects.requireNonNull(one);
        final HashMap<String, Object> urlTplParams = new HashMap<String, Object>(this.pathParameters);
        urlTplParams.put("one", one);
        return new WithOneItemRequestBuilder(urlTplParams, requestAdapter);
    }
    /**
     * Instantiates a new {@link ParametersRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ParametersRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/parameters", pathParameters);
    }
    /**
     * Instantiates a new {@link ParametersRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ParametersRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/parameters", rawUrl);
    }
}
